Knuth-Morris-Pratt/KnuthMorrisPratt.playground/Contents.swift

```diff
@@ -0,0 +1,122 @@
+//: Playground - noun: a place where people can play
+
+
+func ZetaAlgorithm(ptnr: String) -> [Int]? {
+
+    let pattern = Array(ptnr.characters)
+    let patternLength: Int = pattern.count
+
+    guard patternLength > 0 else {
+        return nil
+    }
+
+    var zeta: [Int] = [Int](repeating: 0, count: patternLength)
+
+    var left: Int = 0
```

Comment:
As per our style guide, leave out the explicit typing as Swift can infer the type from the assigned value.

```
// not preferred
var left: Int = 0

// preferred
var left = 0
```